tests/utils/test_transcript_annotation.py

```diff
@@ -192,3 +193,188 @@ def test_create_tx_annotation_by_region(self, sample_hail_table: hl.Table) -> No
 
         # Verify the result
         assert result == expected_result
+
+
+@pytest.fixture
+def mock_vep_annotated_ht():
+    """Create a mock Hail Table with VEP annotations."""
+    return hl.Table.parallelize(
+        [
+            {
+                "locus": hl.Locus("1", 861393, reference_genome="GRCh37"),
+                "alleles": ["G", "A"],
+                "vep": {
+                    "transcript_consequences": [
+                        {
+                            "gene_id": "ENSG00000187634",
+                            "gene_symbol": "SAMD11",
+                            "transcript_id": "ENST00000342066",
+                            "consequence_terms": [
+                                "splice_region_variant",
+                                "synonymous_variant",
+                            ],
+                            "amino_acids": "V",
+                            "biotype": "protein_coding",
+                            "lof": None,
+                            "lof_flags": None,
+                            "canonical": 0,
```

Comment:
it shouldn't matter for testing, but the browser lists ENST00000342066 as the canonical transcript for this gene

Reply:
yeah, I just put random things past what was needed for the gencode filter

Reply:
I wouldn't have put real gene and transcript names if I didn't need to test the gencode filtering